compiler/rustc_hir_analysis/src/astconv/generics.rs

```diff
@@ -590,8 +591,72 @@ pub(crate) fn check_generic_arg_count(
 }
 
 /// Emits an error regarding forbidden type binding associations
-pub fn prohibit_assoc_ty_binding(tcx: TyCtxt<'_>, span: Span) {
-    tcx.sess.emit_err(AssocTypeBindingNotAllowed { span });
+pub fn prohibit_assoc_ty_binding(
+    tcx: TyCtxt<'_>,
+    span: Span,
+    segment: Option<(&hir::PathSegment<'_>, Span)>,
+) {
+    let mut err = tcx.sess.create_err(AssocTypeBindingNotAllowed { span });
```

Comment:
Ideally we'd modify `AssocTypeBindingNotAllowed` to have a new field with the type info and make the text translatable, otherwise with the current approach you could end up with an error half in one language and the other half in English. Adding fields to these errors is relatively straightforward.